packages/flutter_tools/lib/src/runner/flutter_command_runner.dart

```diff
@@ -352,7 +352,15 @@ class FlutterCommandRunner extends CommandRunner<Null> {
         Uri rootUri = flutterUri.resolve('../../..');
         String flutterPath = path.normalize(fs.file(rootUri).absolute.path);
 
-        if (!_compareResolvedPaths(flutterPath, Cache.flutterRoot)) {
+        if (!fs.isDirectorySync(rootUri.path)){
+          // Remove the .packages file that now points to an obsolete SDK location.
+          fs.file(kPackagesFileName).deleteSync();
```

Comment:
This line is going to be confusing when used with commands where we do this automatically (which should be all of them where it matters).

I would replace the entire message with:
```
Warning! This package referenced a Flutter repository that is no longer available.
The offending .packages file in this package has been deleted. The repository
from which the 'flutter' tool is currently executing will be used instead.
  running flutter tool: ...
  previous reference:   ...
This can happen if you deleted or moved your copy of the Flutter repository, or
if it was on a volume that is no longer mounted or has been mounted at a
different location. Please check your system path to verify that you are running
the expected version (run 'flutter --version' to see which flutter is on your path).
```

Also, if you could change the warning of the next branch to read as follows, and have it also delete the .packages file, that would be great. The way things work has changed since we wrote this.

```
Warning! The 'flutter' tool you are currently running is from a different flutter repository
than the one last used by this package. The offending .packages file in this package
has been deleted. The repository from which the 'flutter' tool is currently executing
will be used instead.
  running flutter tool: ...
  previous reference:   ...
This can happen when you have multiple copies of flutter installed. Please check your system
path to verify that you are running the expected version (run 'flutter --version' to see which
flutter is on your path). 
```

Obviously please make sure the line wrapping makes sense (I just dead-reckoned it here) and feel free to make whatever improvements you can come up with. The idea is just to give the user a full explanation of what's going on, and get them on their way as soon as possible.